generators/micronaut/generator.js

```diff
@@ -0,0 +1,283 @@
+      this.fetchFromInstalledJHipster('spring-data-relational/templates'),
+      // For _global_partials_entity_/field_validators file
+      this.fetchFromInstalledJHipster('java/generators/domain/templates'),
+    );
+    await this.dependsOnJHipster(GENERATOR_SERVER);
+    await this.dependsOnJHipster('jhipster:java:build-tool');
+  }
+
+  get [BaseApplicationGenerator.INITIALIZING]() {
+    return this.asInitializingTaskGroup({
+      async initializingTemplateTask() {
+        this.parseJHipsterArguments(command.arguments);
+        this.parseJHipsterOptions(command.options);
+
+        // Disables some prompts by making it configured, it will be shown using `--ask-answered` option.
+        this.jhipsterConfig.reactive = false;
+        this.jhipsterConfig.databaseType = 'sql';
+      },
+    });
+  }
+
+  get [BaseApplicationGenerator.CONFIGURING]() {
+    return this.asConfiguringTaskGroup({
+      async configuringTemplateTask() {
+        this.jhipsterConfig.backendType = 'Micronaut';
+        if (this.jhipsterConfigWithDefaults.authenticationType === 'oauth2') {
+          this.jhipsterConfig.jwtSecretKey = this.jhipsterConfig.jwtSecretKey ?? createBase64Secret(64, this.options.reproducibleTests);
+        }
+      },
+    });
+  }
+
+  get [BaseApplicationGenerator.COMPOSING]() {
+    return this.asComposingTaskGroup({
+      async composing() {
+        const { enableTranslation, databaseType, cacheProvider, skipClient, clientFramework = 'no' } = this.jhipsterConfigWithDefaults;
+
+        await this.composeWithJHipster('jhipster:java:domain');
+        await this.composeWithJHipster('jhipster:java:code-quality');
+        await this.composeWithJHipster('jhipster:java:jib');
+        await this.composeWithJHipster(GENERATOR_DOCKER);
+
+        if (!skipClient && clientFramework !== 'no') {
+          await this.composeWithJHipster('jhipster:java:node');
+        }
+
+        // We don't expose client/server to cli, composing with languages is used for test purposes.
+        if (enableTranslation) {
+          const languagesGenerator = await this.composeWithJHipster(GENERATOR_LANGUAGES);
+          languagesGenerator.writeJavaLanguageFiles = true;
+        }
+        if (databaseType === 'sql') {
+          await this.composeWithJHipster(GENERATOR_LIQUIBASE);
+        }
+        if (['ehcache', 'caffeine', 'hazelcast', 'infinispan', 'memcached', 'redis'].includes(cacheProvider)) {
+          await this.composeWithJHipster('jhipster-micronaut:micronaut-cache');
+        }
+      },
+    });
+  }
+
+  get [BaseApplicationGenerator.PREPARING]() {
+    return this.asPreparingTaskGroup({
+      loadDependencies({ application }) {
+        application.micronautDependencies = { ...mnConstants.versions };
+        this.loadJavaDependenciesFromGradleCatalog(application.micronautDependencies);
+        Object.assign(application.javaDependencies, application.micronautDependencies);
+
+        // Workaound liquibase generator bug
```

Comment:
```suggestion
        // Workaround liquibase generator bug
```
Is there a link to this issue so we'll know if we can remove it in the future?

Reply:
Should be fixed in main already.

Reply:
```suggestion
        // Workaround liquibase generator bug. Drop for generator-jhipster 8.3.1.
```